client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/index.tsx

```diff
@@ -229,27 +240,26 @@ const designSetup: Step = function DesignSetup( { navigation, flow } ) {
 	}
 
 	const handleBackClick = () => {
-		if ( selectedDesign ) {
-			setSelectedDesign( undefined );
-			setIsPreviewingDesign( false );
-		} else {
+		if ( ! selectedDesign || ( showGeneratedDesigns && ! isMobile ) ) {
```

Comment:
I think we could determine we should go back to generated design or previous step as followed:
```js
if ( isPreviewingDesign && ( ! showGeneratedDesigns || isMobile ) {
  setSelectedDesign( undefined );
  setIsPreviewingDesign( false );
  return;
}

if ( isForceStaticDesign ) {
  setIsForceStaticDesigns( false );
  return;
}

goBack();
```

Reply:
Let's go with this approach, and remove it later if it's not how the flow is intended to be.